packages/viewer/src/app.ts

```diff
@@ -41,49 +41,88 @@ async function main(w: Window): Promise<void> {
 
   const document = w.document;
 
-  /**
-   * Parse the DOM looking for LaTeX and <math> elements.
-   * Replaces them with the corresponding rendered images within the given element.
-   * @param {Document} document - The DOM document in which to search for the rendering root.
-   * @param {MutationObserver} observer - Mutation observer to activate or reactivate every time the rendering root changes.
-   */
-  properties.render = async () => {
-    const element: HTMLElement = document.querySelector(properties.element);
-    if (element) {
-      await renderLatex(properties, element);
-      await renderMathML(properties, element);
-    }
-  };
-
   // Initial function to call once document is loaded
   // Renders formulas and sets observer
   const start = async () => {
-    // Check if the viewer is alredy loaded
+    // Check if the viewer is already loaded
     if (w.viewer.isLoaded) return;
 
     w.viewer.isLoaded = true;
 
-    // First render
-    properties.render();
+    // Interaction Observer
+    const mathobserver = new IntersectionObserver(async (entries, observer) => {
+
+      let latexPromises = [];
+
+      const maxMmlPetition = properties.simultaneousmml;
+      let mmlPromises = [];
+      for (const entry of entries) {
+
+        if (entry.isIntersecting && entry.target instanceof MathMLElement && entry.target.matches('math')) {
+          const promise = renderAMathML(properties, entry.target);
```

Comment:
Maybe I am missing something, but: 
Why the renderAMathMl method does not implement the petition guard and renderLatex does? 
Could be possible to implement the guard logic in the same way for both methods? For me is kind of confusing that one does the guard mechanism and the other doesn't.

Reply:
As we talked in private, renderAMathMl and renderLatex have some different behavior, that's why they don't share the same guard mechanism